boa_engine/src/vm/opcode/get/property.rs

```diff
@@ -56,6 +57,23 @@ impl Operation for GetPropertyByValue {
         };
 
         let key = key.to_property_key(context)?;
+
+        // Fast Path
+        if std::ptr::eq(object.vtable(), &ARRAY_EXOTIC_INTERNAL_METHODS) {
```

Comment:
Wouldn't be better to add an `as_array` method to `JsObject` that does this check internally? That should deduplicate this check.